### PR TITLE
HDDS-1502. Add metrics for Ozone Ratis performance.

### PR DESCRIPTION
**Pull request by bshashikant**

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 923 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 457 | trunk passed |
| +1 | compile | 217 | trunk passed |
| +1 | checkstyle | 54 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 867 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 145 | trunk passed |
| 0 | spotbugs | 333 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 580 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 502 | the patch passed |
| +1 | compile | 259 | the patch passed |
| +1 | javac | 259 | the patch passed |
| -0 | checkstyle | 34 | hadoop-hdds: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 722 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 163 | the patch passed |
| +1 | findbugs | 595 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 219 | hadoop-hdds in the patch failed. |
| -1 | unit | 1787 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 37 | The patch does not generate ASF License warnings. |
| | | 7748 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.common.statemachine.commandhandler.TestCloseContainerCommandHandler |
|   | hadoop.ozone.client.rpc.TestBCSID |
|   | hadoop.ozone.container.TestContainerReplication |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestBlockDeletion |
|   | hadoop.ozone.container.common.transport.server.ratis.TestCSMMetrics |
|   | hadoop.ozone.om.TestOzoneManagerHA |
|   | hadoop.ozone.om.TestOmAcls |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestCloseContainerByPipeline |
|   | hadoop.ozone.om.TestOMDbCheckpointServlet |
|   | hadoop.ozone.ozShell.TestOzoneShell |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-833/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/833 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 2c6ce6e248b1 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 0d1d7c8 |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-833/1/artifact/out/diff-checkstyle-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-833/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-833/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-833/1/testReport/ |
| Max. process+thread count | 3780 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/container-service U: hadoop-hdds/container-service |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-833/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/CSMMetrics.java [hunk 1]
```diff
@@ -86,7 +107,7 @@ public long getNumWriteStateMachineOps() {
 
   @VisibleForTesting
   public long getNumReadStateMachineOps() {
-    return numReadStateMachineOps.value();
+    return queryStateMachineOps.value();
```
Comment (lokeshj1703) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/CSMMetrics.java]: Should be numReadStateMachineOps? We need another fn for queryStateMachineOps.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/CSMMetrics.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
 import org.apache.hadoop.metrics2.annotation.Metrics;
 import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
 import org.apache.hadoop.metrics2.lib.MutableCounterLong;
+import org.apache.hadoop.metrics2.lib.MutableRate;
```
[the remark this answers is not in the document]
Comment (bshashikant) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/CSMMetrics.java]: Addressed in the next commit.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/CSMMetrics.java (hunk 1 above)
Comment (bshashikant) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/CSMMetrics.java]: Addressed

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/CSMMetrics.java [hunk 3]
```diff
@@ -38,10 +37,11 @@
...
   private @Metric MutableCounterLong numWriteStateMachineOps;
-  private @Metric MutableCounterLong queryStateMachineOps;
+  private @Metric MutableCounterLong numQueryStateMachineOps;
   private @Metric MutableCounterLong numApplyTransactionOps;
   private @Metric MutableCounterLong numReadStateMachineOps;
   private @Metric MutableCounterLong numBytesWrittenCount;
+  private @Metric MutableCounterLong numBytesCommittedCount;
 
   // Failure Metrics
   private @Metric MutableCounterLong numWriteStateMachineFails;
```
Comment (lokeshj1703) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/CSMMetrics.java]: queryStateMachineFails can be renamed to numQueryStateMachineFails.

**Comment by lokeshj1703** [comment]
@bshashikant  Thanks for working on this! The changes look good to me. I have a minor rename comment. +1 o.w.

**Comment by bshashikant** [comment]
Thanks @lokeshj1703 for the review. I have committed this change to trunk.

**Pull request closed without merging**